torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -414,6 +435,8 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
                 });
           }
         } else if (message.isResponse()) {
+          // Response handling
```

Comment:
I think there are also some exceptions that can happen here (in response handling).  For example with the changes I added in https://github.com/pytorch/pytorch/pull/31381, a runtime error was being raised in `future->markCompleted` and the program would just hang, which made it a lot harder to debug. Catching the exception there could improve the debugging experience.

Reply:
Good catch, I'll cover the markCompleted()/setError() section below where it's possible that the  callback might accidentally throw.